stub_uploader/metadata.py

```diff
@@ -208,8 +208,8 @@ def verify_external_req(
     # TODO: PyPI doesn't seem to have version specific requires_dist. This does mean we can be
     # broken by new releases of upstream packages, even if they do not match the version spec we
     # have for the upstream distribution.
-    if req.name not in [
-        Requirement(r).name for r in (data["info"].get("requires_dist") or [])
+    if req.name.lower() not in [
+        Requirement(r).name.lower() for r in (data["info"].get("requires_dist") or [])
```

Comment:
Can I assume that the requirement we manually specify is always a canonical name? Or should I normalize both like this?
```suggestion
    if canonical_name(req.name) not in [
        canonical_name(Requirement(r).name) for r in (data["info"].get("requires_dist") or [])
```

Reply:
I guess either we should normalize both or we should test somewhere (probably typeshed) that all reqs are in canonical form.

Reply:
Since it was being called twice (thrice with #88). I figured may as well deduplicate it. And now it's only called once anyway, so may as well use the canonical name.